todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskPresenter.java

```diff
@@ -16,47 +16,52 @@
 
 package com.example.android.architecture.blueprints.todoapp.addedittask;
 
-import static com.google.common.base.Preconditions.checkNotNull;
-
+import android.database.Cursor;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.support.v4.app.LoaderManager;
 import android.support.v4.content.Loader;
 
 import com.example.android.architecture.blueprints.todoapp.data.Task;
-import com.example.android.architecture.blueprints.todoapp.data.source.TaskLoader;
+import com.example.android.architecture.blueprints.todoapp.data.source.LoaderProvider;
 import com.example.android.architecture.blueprints.todoapp.data.source.TasksDataSource;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
 
 /**
  * Listens to user actions from the UI ({@link AddEditTaskFragment}), retrieves the data and updates
  * the UI as required.
  */
 public class AddEditTaskPresenter implements AddEditTaskContract.Presenter,
-        LoaderManager.LoaderCallbacks<Task> {
+        LoaderManager.LoaderCallbacks<Cursor>, TasksDataSource.GetTaskCallback {
 
-    private static final int TASK_QUERY = 2;
+    public final static int EDIT_TASK_LOADER = 3;
 
     @NonNull
-    private final TasksDataSource mTasksRepository;
+    private final LoaderProvider mLoaderProvider;
 
     @NonNull
-    private final AddEditTaskContract.View mAddTaskView;
+    private final LoaderManager mLoaderManager;
 
-    @Nullable
-    private String mTaskId;
+    @NonNull
+    private TasksRepository mTasksRepository;
 
-    private TaskLoader mTaskLoader;
+    @NonNull
+    private AddEditTaskContract.View mAddTaskView;
 
-    private final LoaderManager mLoaderManager;
+    @Nullable
+    private String mTaskId;
 
-    public AddEditTaskPresenter(@Nullable String taskId, @NonNull TasksDataSource tasksRepository,
-            @NonNull AddEditTaskContract.View addTaskView, @NonNull TaskLoader taskLoader,
-            @NonNull LoaderManager loaderManager) {
+    public AddEditTaskPresenter(@Nullable String taskId, @NonNull TasksRepository tasksRepository,
+                                @NonNull AddEditTaskContract.View addTaskView, @NonNull LoaderProvider loaderProvider,
```

Comment:
We need to set the styles correctly, in the meantime can you fix these? (no param alignment)